internal/artifact/source.go

```diff
@@ -56,3 +71,13 @@ func WithNopChecksum(rc io.ReadCloser) Source {
 		ChecksumVerifier: nopChecksumVerifier{},
 	}
 }
+
+type gzipReadCloser struct {
+	*gzip.Reader
```

Comment:
nit: i would not embed this one, it makes the implementation of Close() a bit less clear imo